### PR TITLE
Implement Tree -> RVSDG Translation

### PR DESCRIPTION
**Pull request by oflatt**
This PR implements the last missing piece in the translation code, converting the Tree IR to an RVSDG.
It also runs all the tests (excluding ones involving memory) on the full compiler.

### CODE REVIEW

File: src/rvsdg/from_tree.rs [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+        }
+        Type::TupleT(_) => panic!("Tuple types not supported in RVSDG"),
+        Type::Unknown => panic!("Unknown type in tree_type_to_rvsdg_types"),
+    }
+}
+
+fn convert_func_output_type(ty: Type) -> Option<bril_rs::Type> {
+    match ty {
+        Type::TupleT(inner) => {
+            assert_eq!(
```
Comment (ezrosent) [inline src/rvsdg/from_tree.rs:58]: nit: assert!(inner.is_empty()) ? (and maybe add `{inner:?}` to the assertion string?

File: src/rvsdg/from_tree.rs [hunk 2]
```diff
@@ -0,0 +1,412 @@
...
+        }
+        _ => Some(bril_type_from_type(ty)),
+    }
+}
+
+fn tree_func_to_rvsdg(func: RcExpr, program: &TreeProgram) -> RvsdgFunction {
+    let output_type = func.func_output_ty().expect("Expected function types");
+
+    let Type::TupleT(input_types) = func.func_input_ty().expect("Expected function types") else {
+        panic!("Expected tuple type for inputs in tree_func_to_rvsdg")
```
Comment (ezrosent) [inline src/rvsdg/from_tree.rs:73]: Add debug printing to this message (what type did we get instead?)

File: src/rvsdg/from_tree.rs [hunk 3]
```diff
@@ -0,0 +1,412 @@
...
+        // initial state edge is last argument
+        current_state_edge: Operand::Arg(input_types.len()),
+        // initial arguments are the first n arguments
+        current_args: (0..input_types.len()).map(Operand::Arg).collect(),
+    };
+
+    let converted = converter.convert_expr(func.clone());
+
+    let resulting_state_edge = converter.current_state_edge;
+    drop(converter);
```
Comment (ezrosent) [inline src/rvsdg/from_tree.rs:92]: question: why do we need the drop? Does NLL not handle this for you?

File: src/rvsdg/from_tree.rs [hunk 4]
```diff
@@ -0,0 +1,412 @@
...
+            .into_iter()
+            .map(|ty| RvsdgType::Bril(bril_type_from_type(ty)))
+            .chain(iter::once(RvsdgType::PrintState))
+            .collect(),
+        nodes,
+        // functions return a single value and a state edge
+        // or just a state edge
+        results: match convert_func_output_type(output_type) {
+            Some(func_type) => {
+                assert!(converted.len() == 1, "Expected exactly one result");
```
Comment (ezrosent) [inline src/rvsdg/from_tree.rs:108]: nit: assert_eq! ?

**Review by ezrosent — APPROVED** [review]
Pretty straightforward!

What's the issue with the memory ops?

**Comment by oflatt** [comment]
Nothing wrong with memory ops, I haven't tested it. That's my next task

File: src/rvsdg/from_tree.rs (hunk 3 above)
Comment (oflatt) [reply src/rvsdg/from_tree.rs]: We don't need it, I deleted it.
I was confused at some point.

**Pull request merged**